### ISSUE TITLE
Setting new GPX waypoint: Address search is slow

### ISSUE THREAD

**Issue by daganzdaanda** [osmandapp/Osmand#7691]
When you set a new waypoint while recording a track, before you get to type anything you have to wait until the automatic search for the closest address is finished. This can take a long time. 

I use the waypoints to save short text notes when I'm mapping and AV notes would not fit / are not possible at the moment. So in some way this is what the request #7556 is asking for. I have set up a quick action to set a waypoint, but the long wait for the search to be finished (sometimes more than 30 seconds) does kind of defeat the purpose...
There is [this old comment](https://github.com/osmandapp/Osmand/issues/3402#issuecomment-271634193) about the same issue. I agree - it would be nice if the search was done in the background. Or maybe add an option to disable the whole automatic address search, so you could just start typing away to add a note to the waypoint.

(Osmand 3.4.8 on Android 4.1.2)

**Comment by soniok** [comment]
I've noticed it is much faster to search for address when when adding GPX waypoint in remote area compared to in urban area.

**Comment by scaidermern** [comment]
I wonder if anyone actually needs this address search for GPX waypoints. I'm frequently using waypoints for mapping. The address search isn't necessary for me since I always enter my own text.

**Comment by akkana** [comment]
If it's too much work to add a "don't ever search for address" preference, would it be possible to add a Cancel button on the address search dialog? It's frustrating to have to wait on every waypoint for an address search you know will fail.

**Comment by sonora** [comment]
Exactly. We have a "Skip" button on many or most such dialogs enabling you to skip the address search and carry on (like e.g. when adding a point s a Favorite). Have not checked if it is really missing for the Add GPX Waypoint dialog/process, but if it is, I would argue with Victor's "Nice to have" above and would say this is a valid issue which needs fixing. :)

**Comment by daganzdaanda** [comment]
Hi, I have found a possible workaround for the time being: 

- Make a new quick action for "Add a new GPX Waypoint"
- Enter something in the "Name" field, just a single character will do, so that it is NOT empty by default (It says "Leave the field empty to use the address or the Placename" below
- Activate the switch "Show dialogue..."

Now when you add a new GPX Point, the Name field will have your new default until you change it in the dialogue. No search for an address is started, the GPX Waypoint is added very quickly!

**Comment by sonora** [comment]
I guess this issue may be deprecated because there is a solution and also address search has become faster over the years?
